lib/liquid/context.rb

```diff
@@ -126,6 +135,24 @@ def stack(new_scope = nil)
       @this_stack_used = old_stack_used
     end
 
+    # Creates a new context inheriting resource limits, filters, environment etc.,
+    # but with an isolated scope.
+    def new_isolated_subcontext
+      check_overflow
+
+      Context.build(
+        resource_limits: resource_limits,
+        static_environments: static_environments,
+        static_registers: static_registers
+      ).tap do |subcontext|
+        subcontext.base_scope_depth = base_scope_depth + 1
+        subcontext.exception_renderer = exception_renderer
+        subcontext.add_filters(@filters)
```

Comment:
This is only a suggestion and very weakly held. Performance wise does it help to make this `subcontext.filters = @filters`. Just thinking of the case when this is inside `for` blocks, any gain when building context will help the overall speed.

Reply:
:+1: That would be slightly faster. Will update.